libraries/lib-ipc/internal/BufferedIPCChannel.cpp

```diff
@@ -0,0 +1,80 @@
+#include "BufferedIPCChannel.h"
+
+#include <mutex>
+#include <cstring>
+
+BufferedIPCChannel::BufferedIPCChannel(SOCKET socket, IPCChannelStatusCallback* callback)
+   : mCallback(callback), mSocket(socket)
+{
+   mInputBuffer.resize(2048);
+   mOutputBuffer.reserve(2048);
+
+   callback->OnConnect(this);
```

Comment:
Indeed this assumes callback isn't null, so I would rather have a reference argument then